cpp/src/aztec/honk/pcs/shplonk/shplonk_single.hpp

```diff
@@ -44,16 +44,15 @@ template <typename Params> class SingleBatchOpeningScheme {
             max_poly_size = std::max(max_poly_size, poly.size());
         }
         // Q(X) = ∑ⱼ ρʲ ⋅ ( fⱼ(X) − vⱼ) / ( X − xⱼ )
-        Polynomial Q(max_poly_size, max_poly_size);
-        Polynomial tmp(max_poly_size, max_poly_size);
```

Comment:
I agree for sure - should have left a comment here. I did this because this was the only point in production code that was calling the 'bump_memory' function, which doesn't seem like a desirable behavior. My thought was that this code has not been optimized and there should be a way to simply initialize a poly with sufficient memory then reuse that. (I think that was your intent here but I don't think that was the result). I'll take another look at this and see if there's an easy fix.

Reply:
This is now restored.